tests/cuda_interop/cuda_interop_device_function.cu

```diff
@@ -0,0 +1,135 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright:  (c) 2017 by Codeplay Software LTD. All Rights Reserved.
+//
+*******************************************************************************/
+
+#include "../common/common.h"
+
+#ifdef SYCL_BACKEND_CUDA
+#include "../../util/test_base_cuda.h"
+
+#endif
+
+#define TEST_NAME cuda_interop_device_fucntion
+
+namespace cuda_interop_device_fucntion__ {
+
+using namespace sycl_cts;
+
+template <typename T>
+SYCL_EXTERNAL __device__ T func(T *i) {
+  return *i + 1;
+}
+
+/** check device fucntion can be called with the backend interop type `T *`
+converted from an `accessor`
+  */
+template <typename T>
+void test_device_function_buffer(sycl::queue &queue,
+                                 sycl_cts::util::logger &log,
+                                 const std::string &typeName) {
+  size_t constexpr size = 1;
+  T data[size];
+  data[0] = 0;
+  bool result[size] = {false};
+  {
+    sycl::buffer<T> buff(data, sycl::range<1>(size));
+    sycl::buffer<bool> result_buf(result, sycl::range<1>(size));
+
+    queue.submit([&](sycl::handler &cgh) {
+      auto acc = buff.template get_access<sycl::access::mode::read>(cgh);
+      auto result_acc = result_buf.get_access<sycl::access::mode::write>(cgh);
+
+      cgh.single_task([=]() {
+        result_acc[0] =
+            func<T>(sycl::get_native<sycl::backend::cuda>(acc)) == 1;
+      });
+    });
+  }
+
+  if (!result[0]) {
+    log.note(
+        "Test for CUDA device function interop with accessor failed for \"" +
+        typeName + "\" type");
+  }
+  assert(result[0]);
+}
+
+/** check device fucntion can be called with the backend interop type `T *`
+converted from an `local_accessor`
+  */
+template <typename T>
+void test_device_function_local(sycl::queue &queue, sycl_cts::util::logger &log,
+                                const std::string &typeName) {
+  size_t constexpr size = 1;
+  bool result[size] = {false};
+  {
+    sycl::buffer<bool> result_buf(result, sycl::range<1>(size));
+
+    queue.submit([&](sycl::handler &cgh) {
+      sycl::local_accessor<T> acc(size, cgh);
+      auto result_acc = result_buf.get_access<sycl::access::mode::write>(cgh);
```

Comment:
As `local_accessor` allocates device local memory by itself [[spec](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:accessor.local)], I have a feeling that this will not work.

Reply:
So I do not quite understand the comment. Could you clarify what will not work?

Reply:
Oh, I'm sorry, for some reason I thought that you assigned `result_buf.get_access()` to the `local_accessor` instance.

I have no more concerns about this code :)